server/storage_provider.go

```diff
@@ -14,87 +14,36 @@
 
 package server
 
-import (
-	"flag"
-	"fmt"
-	"sync"
+import "github.com/google/trillian/storage"
 
-	"github.com/google/trillian/monitoring"
-	"github.com/google/trillian/storage"
-)
+// TODO(pavelkalinnikov): This file contains type/function aliases for backward
+// compatibility purposes. It will be removed with the next major version bump.
 
-// NewStorageProviderFunc is the signature of a function which can be registered
-// to provide instances of storage providers.
-type NewStorageProviderFunc func(monitoring.MetricFactory) (StorageProvider, error)
-
-var (
-	storageSystem = flag.String("storage_system", "mysql", fmt.Sprintf("Storage system to use. One of: %v", storageProviders()))
+// NewStorageProviderFunc is the signature of a function which can be
+// registered to provide instances of storage providers.
+//
+// Deprecated: storage.NewProviderFunc must be used directly.
```

Comment:
nit: Instead of "must", rephrase as "should" - it'll still work with the aliases (for now), so it's not strictly a "must".